### PR TITLE
Pin Pytorch 2.2 in IPEX Tests

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

Until a compatible ipex is released.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
     assert False, "Error: Could not open '%s' due %s\n" % (filepath, error)
 
 INSTALL_REQUIRE = [
-    "torch>=1.11",
+    "torch>=1.11,<2.3",
```
Comment (echarlaix) [inline setup.py:30]: I think it make sense to pin the torch version for IPEX tests only, the IPEX version should always match the torch version (it's not related to optimum-intel)

**Comment by helena-intel** [comment]
OpenVINO tests pass with PyTorch 2.3 ( https://github.com/huggingface/optimum-intel/actions/runs/8821417491/job/24217231927 ). @eaidova do you know of any issues with OpenVINO and PyTorch 2.3? If there are no issues with OpenVINO the PyTorch version should be limited only in the INC/IPEX extra.

File: setup.py (hunk 1 above)
Comment (IlyasMoutawwakil) [reply setup.py:30]: thanks makes sense

**Review by echarlaix — APPROVED** [review]
Thanks @IlyasMoutawwakil !

**Pull request merged**